File: src/GLFMmatlab/generate_toy_images.m
function [data,gT] = generate_toy_images(N,s2x)
    % Inputs:
    %   s2x: observation noise
    %     N: number of samples
    %
    % Outputs:
    %  data: data structure with X: N*D obs matrix and C: 1*D datatype
    %  string vector
    %
    Btrue = 2* [0,1.0,0,0,0,0,  1,1,1,0,0,0, 0,1,0,0,0,0, ...
        0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0; ...
        0,0.0,0,1,1,1,  0,0,0,1,0,1, 0,0,0,1,1,1, ...
        0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0; ...
        0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0, ...
        1,0,0,0,0,0, 1,1,0,0,0,0, 1,1,1,0,0,0; ...
        0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0, ...
        0,0,0,1,1,1, 0,0,0,0,1,0, 0,0,0,0,1,0];
    
    K = size(Btrue,1); % number of latent features
    D = size(Btrue,2); % number of dimensions
    
    Ztrue = rand(N,K) < 0.5;
    
    data.X = sqrt(s2x) * randn(N,D) + ( Ztrue * Btrue );
    data.C = repmat('g',1,D);
    
    gT.B = Btrue;
    gT.Z = Ztrue;
    
end

File: src/GLFMR/generate_toy_images.R

#' @param s2x: observation noise
#' @param N: number of samples
#' @return data: data structure with X: N*D obs matrix and
#' @return  C: 1*D datatype string vector
%

 generate_toy_images<-function(N,s2x){
Btrue<-matrix(c(c(0,1.0,0,0,0,0,1,1,1,0,0,0, 0,1,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0),
                c(0,0.0,0,1,1,1,0,0,0,1,0,1,0,0,0,1,1,1,0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0),
              c(0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,1,0,0,0,0, 1,1,1,0,0,0),
              c(0,0,0,0,0,0, 0,0,0,0,0,0, 0,0,0,0,0,0,0,0,0,1,1,1, 0,0,0,0,1,0, 0,0,0,0,1,0)),
              nrow=4,ncol=36,byrow=TRUE)
K <- dim(Btrue)[1] # number of latent features
D <- dim(Btrue)[2] # number of dimensions
m0<-matrix(0,nrow=N,ncol=K)
Ztrue <- apply(m0, c(1,2), function(x) sample(c(0,1),1,prob=c(0.2,0.8)))
norm_mat<-matrix(rnorm(N*D),nrow=N,ncol=D,byrow=TRUE)
X <- sqrt(s2x) * norm_mat +  Ztrue %*% Btrue 
C <- rep('g',D)
data<-list("X"=X,"C"=C)
gT<-list("B"=Btrue,"Z"=Ztrue)
return(list(data,gT))
}

